bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/servlets/ChildrenServlet.java

```diff
@@ -0,0 +1,108 @@
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.sandbox.internal.servlets;
+
+import java.io.IOException;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(
+    service = Servlet.class,
+    property = {
+        "sling.servlet.methods=POST",
+        "sling.servlet.resourceTypes=sling/servlet/default",
+        "sling.servlet.selectors=" + ChildrenServlet.SELECTOR,
+        "sling.servlet.extensions=html"
+    }
+)
+public class ChildrenServlet extends SlingAllMethodsServlet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ChildrenServlet.class);
+
+    protected static final String SELECTOR = "children";
+    private static final String PARAM_DELETED_CHILDREN = "deletedChildren";
+    private static final String PARAM_OREDERED_CHILDREN = "orderedChildren";
+
+    @Override
+    protected void doPost(SlingHttpServletRequest request,
+                          final SlingHttpServletResponse response)
+        throws ServletException, IOException {
+
+        ResourceResolver resolver = request.getResourceResolver();
+        Resource container = request.getResource();
+
+        // Remove the child items
+        String[] deletedChildrenNames = request.getParameterValues(PARAM_DELETED_CHILDREN);
+        if (deletedChildrenNames != null && deletedChildrenNames.length > 0) {
+            for (String childName: deletedChildrenNames) {
+                Resource child = container.getChild(childName);
+                if (child != null) {
+                    resolver.delete(child);
+                    resolver.commit();
+                }
+            }
+        }
+
+        // Re-order the child items
+        String[] orderedChildrenNames = request.getParameterValues(PARAM_OREDERED_CHILDREN);
+        if (orderedChildrenNames != null && orderedChildrenNames.length > 0) {
+            final Node containerNode = container.adaptTo(Node.class);
+            if (containerNode != null) {
+
+                // Create the items if they don't exist
+                for (int i = 0; i < orderedChildrenNames.length; i++) {
+                    try {
+                        if (!containerNode.hasNode(orderedChildrenNames[i])) {
+                            containerNode.addNode(orderedChildrenNames[i]);
+                        }
+                    } catch (RepositoryException e) {
```

Comment:
Catching the exception here causes the loss of the node! We should not catch any exception in here, but let it terminate the complete operation.

Reply:
@joerghoh Would you rather place the complete reordering operation within the try-catch block to make sure the whole operation fails if a `RepositoryException` is thrown?

Reply:
Yes, all or nothing would be better. But that would also means a single commit.

Reply:
@joerghoh It has been addressed in https://github.com/Adobe-Marketing-Cloud/aem-core-wcm-components/commit/20ed2ff11112adebb970830c333e23ccb8f227fb